base/commands/map/map_destroy.go

```diff
@@ -0,0 +1,65 @@
+package _map
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hazelcast/hazelcast-go-client"
+
+	"github.com/hazelcast/hazelcast-commandline-client/clc"
+	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/plug"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/prompt"
+)
+
+const (
+	mapDestroyFlagAutoYes = "yes"
+)
+
+type MapDestroyCommand struct{}
+
+func (mc *MapDestroyCommand) Init(cc plug.InitContext) error {
+	help := "Destroy a Map"
+	cc.SetCommandHelp(help, help)
+	cc.AddBoolFlag(mapDestroyFlagAutoYes, "", false, false, "skip interactive approval")
+	cc.SetCommandUsage("destroy")
+	return nil
+}
+
+func (mc *MapDestroyCommand) Exec(ctx context.Context, ec plug.ExecContext) error {
+	mv, err := ec.Props().GetBlocking(mapPropertyName)
+	if err != nil {
+		return err
+	}
+
+	autoYes := ec.Props().GetBool(mapDestroyFlagAutoYes)
+	if !autoYes {
+		prompt := prompt.NewPrompter(ec.Stdin(), ec.Stdout())
+		yes, err := prompt.YesNoPrompt("Map will be deleted irreversibly, do you agree?")
+		if err != nil {
+			return err
```

Comment:
What kind of error we get here?

Reply:
When user gives a wrong input or any error when getting the user input.

Reply:
Would it make sense to assume the answer is "no" if there's an error?

Reply:
The end result is like the answer was no and an error could help us find out if there was a real problem getting user's input.

Reply:
How about assuming the answer is "no" but logging the error?